### PR TITLE
security/acme-client: add support to multiple dns api providers

### PR DESCRIPTION
**Pull request by DonMcCoy**
Add support to other DNS APIs:

autoDNS (InternetX)
Azure DNS
DirectAdmin 
DreamHost DNS
INWX 
KingHost DNS
Knot (knsupdate) DNS
Namesilo.com DNS
selectel.com / selectel.ru
Servercow 
UnoEuro
Zilore DNS
zonomi.com

### CODE REVIEW

File: security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml [hunk 1]
```diff
@@ -199,6 +244,22 @@
...
+    <field>
+        <id>validation.dns_da_key</id>
+        <label>API Key</label>
+        <type>text</type>
+    </field>
+    <field>
+        <id>validation.dns_da_insecure</id>
+        <label>Disable SSL Verification</label>
+        <type>checkbox</type>
+        <help><![CDATA[Uncheck this box if you have a valid SSL certificate for your DirectAdmin installation]]></help>
```
Comment (fabianfrz) [inline security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml]: you probably don't need CDATA here -> no entities

File: security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml [hunk 2]
```diff
@@ -390,6 +477,39 @@
...
+    <field>
+        <label>Knot</label>
+        <type>header</type>
+        <style>table_dns table_dns_knot</style>
+    </field>
+    <field>
+        <id>validation.dns_knot_server</id>
+        <label>Server</label>
+        <type>text</type>
+        <help><![CDATA[Knot server address, like: dns.example.com]]></help>
```
Comment (fabianfrz) [inline security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml]: no cdata needed

File: security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml [hunk 3]
```diff
@@ -390,6 +477,39 @@
...
+    </field>
+    <field>
+        <id>validation.dns_knot_key</id>
+        <label>Key</label>
+        <type>text</type>
+        <help><![CDATA[Specify the location of the generated TSIG Key inside the TSIG file using grep and cut, example: grep \# /etc/knot/acme.key | cut -d' ' -f2 ]]></help>
```
Comment (fabianfrz) [inline security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml]: very likely also here

**Comment by DonMcCoy** [comment]
Yes you're totally right, as the text doesn't contain any markup chars there's totally no point of not parsing the section, I will correct it soon

**Comment by DonMcCoy** [comment]
@fabianfrz Corrected with commit https://github.com/opnsense/plugins/pull/712/commits/c5b049cd9a3f8b6966ae32efe220696f0ddaad62, you can continue reviewing, thanks

**Comment by fraenki** [comment]
Merged, thanks! :)

**Pull request merged**